### ISSUE TITLE
Incorrect determination from `ActiveRecord::Persistence#previously_new_record?` after create-then-destroy

### ISSUE THREAD

**Issue by eizengan** [rails/rails#48794]
### Steps to reproduce

- create a record
- destroy the record

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem "rails", github: "rails/rails", branch: "main"
  gem "sqlite3"
end

require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :records, force: true do |t|
    t.string :text
  end
end

class Record < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_previously_new_record
    destroyed_record = Record.create(text: "a record we will destroy")
    updated_record = Record.create(text: "a record we will update")

    destroyed_record.destroy!
    updated_record.update!(text: "a record we have updated")

    assert_equal updated_record.previously_new_record?, destroyed_record.previously_new_record?
  end
end

```

### Expected behavior
`ActiveRecord::Persistence#previously_new_record?` should be false after create and subsequent destroy, as it would be e.g. after create and subsequent update.

### Actual behavior
`ActiveRecord::Persistence#previously_new_record?` is true after create and subsequent destroy.

This causes callbacks such as the following to unexpectedly fire twice for the destroyed record during the test above:

```ruby
on_commit :do_something_with_the_record, if: :previously_new_record?
```

There is, however, a semantic argument to be made about what "previously new" means. Depending on interpretation, this behavior could be intentional. 

### System configuration
**Rails version**: 7.0.6

**Ruby version**: 3.2.2

**Comment by akhilgkrishnan** [comment]
Fixes by #48796.

**Comment by skipkayhil** [comment]
> Fixes by #48796. So we can close this issue.

Nice catch, I'll ask if it should be backported tomorrow but I'll close the issue since its fixed on main
